core-bundle/src/EventListener/DataContainer/PageCallbackListener.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\DataContainer;
+use Contao\PageModel;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+class PageCallbackListener implements ServiceAnnotationInterface
```

Comment:
we should use a more descriptive class name (instead of the method name). Like `PageRedirectListener`

Reply:
Yes, if we decide that we do _not_ collect all `tl_page` callbacks into a single class in the future.

Reply:
They should not be all in the same class because they use different dependencies. But maybe we wanna have a subnamespace for each container? So `DataContainer\Page\XListener` etc.

Reply:
Well, Looking at the current `tl_page` callbacks, some callbacks do share dependencies. e.g. database, session & security. Some methods are also shared. But otherwise I have nothing against using a callable called `Contao\CoreBundle\EventListener\DataContainer\Page\AdjustAutoforwardSubpaletteListener` 👍

Reply:
Please do not make it more complicated then necessary though. 🙈

Reply:
@leofeyer so no sub name space? ;). This structure is also what I currently use in projects with lots of DCA callbacks for different tables.